### PR TITLE
Adding walletconnect, walletlink (coinbase wallet), and 0xFrame to web3modal

### PR DESCRIPTION
**Pull request by karmabadger**
Adding walletconnect, walletlink (coinbase wallet), and 0xFrame to web3modal.
This is a bit of of a breaking change.
The first commit is trivial while the rest would require adding react-rewire as there are some node polyfills required.

an infura id to be added would be required (probably best to add as an environment variable though.

Thanks

### CODE REVIEW

File: contracts/package.json [hunk 1]
```diff
@@ -62,5 +62,10 @@
         "typechain": "^5.1.2",
         "typescript": "^4.5.5"
     },
-    "dependencies": {}
+    "dependencies": {
+        "@walletconnect/web3-provider": "^1.7.3",
+        "eth-provider": "^0.10.0",
+        "react-app-rewired": "^2.2.1",
+        "walletlink": "^2.5.0"
+    }
```
Comment (reisr3) [inline contracts/package.json:70]: These got added to the `/contracts` package.json - they should be in the `/frontend` one.

File: frontend/.env.example [hunk 2]
```diff
@@ -16,4 +16,7 @@ REACT_APP_API="http://localhost:4000"
 # Fortmatic wallet - not used yet
 REACT_APP_FORTMATIC_KEY=""
 
-REACT_APP_PAYMENT_TOKEN_NAME="ETH"
\ No newline at end of file
+REACT_APP_PAYMENT_TOKEN_NAME="ETH"
+
+
+INFURA_PROJECT_ID="9aa3d95b3bc440fa88ea12eaa4456161"
```
Comment (reisr3) [inline frontend/.env.example:22]: Needs to be prefixed with `REACT_APP_` or it won't be read by CRA.

File: frontend/config-overrides.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  };
+  config.plugins.push(
+    new webpack.ProvidePlugin({
+      process: 'process/browser',
+      Buffer: ['buffer', 'Buffer'],
+    })
+  );
+
+  return config;
+};
```
Comment (reisr3) [inline frontend/config-overrides.js:21]: Why are we doing this?

File: frontend/src/components/wallet/useWalletProvider.ts [hunk 4]
```diff
@@ -9,19 +11,44 @@ import { useEffect, useMemo, useReducer, useState } from 'react';
...
 import { walletReducer } from './WalletReducer';
 
+const ethProvider = require('eth-provider');
 const fortmaticNetworkOptions = {
   rpcUrl: 'https://rpc-mainnet.maticvigil.com',
   chainId: 137,
 };
 
+// this should be replaced by an env variable. I provided a dummy one for now
+const infuraId = 'f8f8f8f8f8f8f8f8f8f8f8f8f8f8f8f8';
```
Comment (reisr3) [inline frontend/src/components/wallet/useWalletProvider.ts:21]: Use the environment variable.

File: frontend/src/components/wallet/useWalletProvider.ts [hunk 5]
```diff
@@ -9,19 +11,44 @@ import { useEffect, useMemo, useReducer, useState } from 'react';
...
+  walletconnect: {
+    package: WalletConnectProvider,
+    options: {
+      infuraId: infuraId,
+    },
+  },
+  walletlink: {
+    package: WalletLink, // Required
+    options: {
+      appName: 'My Awesome App', // Required
```
Comment (reisr3) [inline frontend/src/components/wallet/useWalletProvider.ts:34]: Change to `EngineerDAO`

File: frontend/src/components/wallet/useWalletProvider.ts [hunk 6]
```diff
@@ -9,19 +11,44 @@ import { useEffect, useMemo, useReducer, useState } from 'react';
...
+      infuraId: infuraId, // Required unless you provide a JSON RPC url; see `rpc` below
+      rpc: '', // Optional if `infuraId` is provided; otherwise it's required
+      chainId: 1, // Optional. It defaults to 1 if not provided
```
Comment (reisr3) [inline frontend/src/components/wallet/useWalletProvider.ts:37]: Can you make an env var for this chainId and the one above as well?

File: frontend/config-overrides.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
+/* config-overrides.js */
+const webpack = require('webpack');
+module.exports = function override(config, env) {
+  //do stuff with the webpack config...
```
Comment (reisr3) [inline frontend/config-overrides.js:4]: Can we remove this comment?

File: frontend/config-overrides.js (hunk 3 above)
Comment (reisr3) [reply frontend/config-overrides.js:21]: Answered in Discord: we have to polyfill this stuff for webpack 5.

**Review by reisr3 — CHANGES_REQUESTED** [review]
Looking good! Just a few changes.

File: contracts/package.json (hunk 1 above)
Comment (karmabadger) [reply contracts/package.json:70]: oh fook my bad LOL

Comment (karmabadger) [reply contracts/package.json:70]: removed

File: frontend/.env.example (hunk 2 above)
Comment (karmabadger) [reply frontend/.env.example:22]: added

File: frontend/src/components/wallet/useWalletProvider.ts (hunk 4 above)
Comment (karmabadger) [reply frontend/src/components/wallet/useWalletProvider.ts:21]: added

File: frontend/src/components/wallet/useWalletProvider.ts (hunk 5 above)
Comment (karmabadger) [reply frontend/src/components/wallet/useWalletProvider.ts:34]: changed

**Comment by karmabadger** [comment]
lol I just discovered that I can click on "resolve"
ok this will be combined with https://github.com/engineer-dao/platform/pull/117
in a single merge

**Pull request closed without merging**